features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5_SDK13/source/btle/btle_security_pm.cpp

```diff
@@ -434,4 +434,59 @@ btle_generateResolvableAddress(const ble_gap_irk_t &irk, ble_gap_addr_t &address
     /* Calculate the hash and store it in the top half of the address */
     ah(irk.irk, &address.addr[BLE_GAP_ADDR_LEN - 3], address.addr);
 }
+
+
+#if (NRF_SD_BLE_API_VERSION >= 3)
+ble_error_t btle_getAddressesFromBondTable(Gap::Whitelist_t &addrList)
+{
+    pm_peer_id_t           peer_id;
+    ret_code_t             ret;
+    pm_peer_data_bonding_t bond_data;
+
+    addrList.size = 0;
+    peer_id       = pm_next_peer_id_get(PM_PEER_ID_INVALID);
+
+    /**
+     * Fill addresses list:
+     * Copy addresses from bond table, or
+     * for every private resolvable address in the bond table check if
+     * there is maching address in th provided whitelist and copy resolvable address.
```

Comment:
Maybe I'm missing something, but I feel a bit confused with this comment. I can't see below any checking _if there is a matching address_. And what is _the provided whitelist_?